### PR TITLE
fix(amplify-codegen-appsync-model-plugin): use LazyCollection (JS/TS)

### PR DESCRIPTION
**Pull request by yuth**
Connected models to use LazyCollection in the generated Java Script and TypeScript code

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-codegen-appsync-model-plugin/package.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
     "graphql-toolkit": "^0.5.17",
     "pluralize": "^8.0.0",
     "strip-indent": "^3.0.0",
-    "ts-dedent": "^1.1.0"
+    "ts-dedent": "^1.1.0",
+    "chalk": "^3.0.0"
```
Comment (attilah) [inline packages/amplify-codegen-appsync-model-plugin/package.json]: Please use ^2.4.2 as in other packages.

File: packages/amplify-codegen-appsync-model-plugin/src/utils/warn.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import chalk from 'chalk';
+const renderedWarnings: string[] = [];
```
Comment (attilah) [inline packages/amplify-codegen-appsync-model-plugin/src/utils/warn.ts]: nit: ```printedWarnings``` as method is called ```printWarning```

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-visitor.ts [hunk 3]
```diff
@@ -293,13 +294,13 @@ export class AppSyncModelVisitor<
     } else if (this.isEnumType(field)) {
       typeNameStr = this.getEnumName(this.enumMap[typeName]);
     } else {
-      throw new Error(`Unknown type ${typeName} for field ${field.name}`);
+      throw new Error(`Unknown type ${typeName} for field ${field.name}. Did you miss adding @model directive?`);
```
Comment (attilah) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-visitor.ts]: What about: "Did you forget to add the @model directive"

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-visitor.ts [hunk 4]
```diff
@@ -404,8 +405,8 @@ export class AppSyncModelVisitor<
         const fieldType = field.type;
         const connectionInfo = field.connectionInfo;
         if (modelTypes.includes(fieldType) && connectionInfo === undefined) {
-          console.warn(
-            `Model ${model.name} has field ${field.name} of type ${field.type} but its not connected. Please add @connection directive to connect them`
+          printWarning(
+            `Model ${model.name} has field ${field.name} of type ${field.type} but its not connected. Add @connection directive if want to connect them.`
```
Comment (attilah) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-visitor.ts]: "Add a @connection..."

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts [hunk 5]
```diff
@@ -175,7 +175,7 @@ export class AppSyncSwiftVisitor extends AppSyncModelVisitor {
 
     return result.join('\n');
   }
-  protected getListType(typeStr: string): string {
+  protected getListType(typeStr: string, field: CodeGenField): string {
```
Comment (attilah) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts]: Other ```getListType``` methods are returning a type that is either ```List<>``` or ```[]``` is it intentional here that just returning the incoming ```typeStr``` parameter? also no need for string templates as nothing is added to the incoming param.

Comment (yuth) [reply packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts]: Yes this is intentional as the declaration block knows how to handle if its a list

**Pull request merged**